dbt/adapters/duckdb/plugins/motherduck.py

```diff
@@ -8,40 +10,73 @@
+            for attachment in self.creds.motherduck_attach:
+                parsed = urlparse(attachment.path)
+                qs = parse_qs(parsed.query)
+                for KEY in MOTHERDUCK_CONFIG_OPTIONS:
+                    value = qs.get(KEY)
+                    if value:
+                        conn.execute(f"SET {KEY} = '{value[0]}'")
+            # If config options are specified via plugin config, set them here
+            if self._config:
+                conn.execute(f"SET {MOTHERDUCK_TOKEN} = '{self._token}'")
+            elif self.creds.settings:
+                if MOTHERDUCK_TOKEN in self.creds.settings:
+                    token = self.creds.settings.pop(MOTHERDUCK_TOKEN)
+                    conn.execute(f"SET {MOTHERDUCK_TOKEN} = '{token}'")
 
     @staticmethod
-    def token_from_config(creds: DuckDBCredentials) -> str:
+    def token_from_config(config: Dict[str, Any]) -> str:
         """Load the token from the MotherDuck plugin config
         If not specified, this returns an empty string
 
         :param str: MotherDuck token
         """
-        plugins = creds.plugins or []
-        for plugin in plugins:
-            if plugin.config:
-                token = plugin.config.get("token") or ""
-                return str(token)
+        if (
+            TOKEN in config
+            or TOKEN.upper() in config
+            or MOTHERDUCK_TOKEN in config
+            or MOTHERDUCK_TOKEN.upper() in config
+        ):
+            token = (
+                config.get(TOKEN)
+                or config.get(TOKEN.upper())
+                or config.get(MOTHERDUCK_TOKEN)
+                or config.get(MOTHERDUCK_TOKEN.upper())
+            )
+            return str(token)
         return ""
 
     def update_connection_config(self, creds: DuckDBCredentials, config: Dict[str, Any]):
         user_agent = f"dbt/{__version__} dbt-duckdb/{__plugin_version__}"
-        if "custom_user_agent" in config:
-            user_agent = f"{user_agent} {config['custom_user_agent']}"
+        if CUSTOM_USER_AGENT in config:
+            user_agent = f"{user_agent} {config[CUSTOM_USER_AGENT]}"
         settings: Dict[str, Any] = creds.settings or {}
-        if "custom_user_agent" in settings:
-            user_agent = f"{user_agent} {settings.pop('custom_user_agent')}"
+        if CUSTOM_USER_AGENT in settings:
+            user_agent = f"{user_agent} {settings.pop(CUSTOM_USER_AGENT)}"
 
-        config["custom_user_agent"] = user_agent
+        config[CUSTOM_USER_AGENT] = user_agent
 
-        # If a user specified the token via the plugin config,
-        # pass it to the config kwarg in duckdb.connect
-        token = self.token_from_config(creds)
-        if token != "":
-            config["motherduck_token"] = token
+        # If a user specified MotherDuck config options via the plugin config,
+        # pass it to the config kwarg in duckdb.connect.
+        if not creds.is_motherduck_attach:
+            if self._token:
```

Comment:
I think this can be simplified to

```suggestion
        if not creds.is_motherduck_attach and self._token:
```